app/code/Magento/Sales/Block/Adminhtml/Order/View.php

```diff
@@ -451,7 +459,7 @@ public function getBackUrl()
             return $this->getOrder()->getBackUrl();
         }
 
-        return $this->getUrl('sales/*/');
+        return $this->getUrl('sales/*/',['back'=>true]);
```

Comment:
I don't think it's worth it to introduce a new fake parameter for removing of the existing unneeded parameter.

What do you think about retrieving the URL directly from the URL builder to avoid adding `order_id` param in `getUrl` method:
```suggestion
        return $this->_urlBuilder->getUrl('sales/*/');
```